packages/blade/src/components/Checkbox/useCheckbox.ts

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import type { GestureResponderEvent } from 'react-native';
+import { useControllableState } from '../../hooks/useControllable';
+import { useId } from '../../hooks/useId';
+import { getPlatformType, makeAccessible } from '../../utils';
+import type { CheckboxProps } from './Checkbox';
+
+type UseCheckboxProps = Pick<
+  CheckboxProps,
+  | 'isChecked'
+  | 'defaultChecked'
+  | 'isDisabled'
+  | 'isIndeterminate'
+  | 'isRequired'
+  | 'onChange'
+  | 'name'
+  | 'value'
+> & { hasError?: boolean };
```

Comment:
why do you need hasError here? the api decisions doesn't has anything WRT hasError right?

Reply:
This is useCheckbox not the public API.

Even if I pass validationState I'll need to convert it to boolean `validationState === 'error'` so might as well just accept hasError.